app/assets/javascripts/index/contextmenu.js

```diff
@@ -64,6 +62,10 @@ OSM.initializeContextMenu = function (map) {
     else map.contextmenu.enable();
   });
 
+  function stringifyArgs(args) {
+    return new URLSearchParams(args).toString();
+  }
+
```

Comment:
Is this function required? You can replace 
```js
OSM.router.route("/directions?" + stringifyArgs({
...
```
with
```js
OSM.router.route("/directions?" + new URLSearchParams({
...
```
and this will work because string coercion runs `toString`.

Reply:
Thanks for point that out - hopefully I've fixed all those now.